symengine/dense_matrix.cpp

```diff
@@ -1613,6 +1632,30 @@ void cross(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
 
 RCP<const Set> eigen_values(const DenseMatrix &A)
 {
+    unsigned n = A.nrows();
+    bool is_upper = true, is_lower = true;
+    for (unsigned i = 0; i < n; ++i) {
+        for (unsigned j = 0; j < n; ++j) {
+            if (i > j and not eq(*(A.get(i, j)), *zero)) {
+                is_lower = false;
+                break;
+            } else if (i < j and not eq(*(A.get(i, j)), *zero)) {
+                is_upper = false;
+                break;
```

Comment:
Breaking here would interfere with checking `is_lower`.

Reply:
I suggest making two separate checks. One for `is_lower` and one for `is_upper`

Reply:
I got it, Thanks for your guidance :)